src/orange/OrangeTypes.hh

```diff
@@ -436,6 +424,67 @@ CELER_CONSTEXPR_FUNCTION Sense to_sense(SignedSense s)
     return Sense(static_cast<int>(s) >= 0);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Wrapper for a sense value which associated metadata.
+ */
+class SenseMeta
+{
+    /*!
+     * Bitset indices for the sense metadata.
+     */
+    enum SenseFlags : unsigned char
+    {
+        cache,  //!< Sense is cached
+        sense,  //!< Sense value
```

Comment:
Maybe it doesn't matter, but wouldn't we want `sense` to be the first bit? That way we could interpret a vector of senses as a vector of SenseMeta.